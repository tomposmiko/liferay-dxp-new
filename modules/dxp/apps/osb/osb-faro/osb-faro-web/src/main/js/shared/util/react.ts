import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
