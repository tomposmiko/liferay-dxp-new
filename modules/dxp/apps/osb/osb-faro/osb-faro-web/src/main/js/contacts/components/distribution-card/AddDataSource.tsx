import Card from 'shared/components/Card';
import ClayLink from '@clayui/link';
import NoResultsDisplay from 'shared/components/NoResultsDisplay';
import React from 'react';
import {Routes, toRoute} from 'shared/util/router';

interface IAddDataSourceProps {
	groupId: string;
}

const AddDataSource: React.FC<IAddDataSourceProps> = ({groupId}) => (
	<>
		<Card.Header>
			<Card.Title>{Liferay.Language.get('distribution')}</Card.Title>
		</Card.Header>

		<Card.Body>
			<NoResultsDisplay
				description={Liferay.Language.get(
					'convert-and-enrich-anonymous-individuals-to-known-individuals-with-attributes-from-another-data-source'
				)}
				primary
				title=''
			>
				<ClayLink
					button
					className='button-root'
					displayType='primary'
					href={toRoute(Routes.SETTINGS_ADD_DATA_SOURCE, {groupId})}
				>
					{Liferay.Language.get('add-data-source')}
				</ClayLink>
			</NoResultsDisplay>
		</Card.Body>
	</>
);

export default AddDataSource;
