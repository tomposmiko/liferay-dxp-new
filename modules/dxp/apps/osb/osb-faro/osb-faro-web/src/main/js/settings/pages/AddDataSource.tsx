import * as breadcrumbs from 'shared/util/breadcrumbs';
import BasePage from 'settings/components/BasePage';
import getCN from 'classnames';
import React from 'react';
import SelectDataSource from '../components/SelectDataSource';
import {close, modalTypes, open} from 'shared/actions/modals';
import {compose, withAdminPermission} from 'shared/hoc';
import {connect} from 'react-redux';
import {ENABLE_CSVFILE} from 'shared/util/constants';
import {ENABLE_SALESFORCE} from 'shared/util/constants';
import {Routes, toRoute} from 'shared/util/router';

interface IAddDataSourceProps extends React.HTMLAttributes<HTMLDivElement> {
	close: () => void;
	groupId: string;
	open: (
		modalType: string,
		{groupId, onClose}: {groupId: string; onClose: () => void}
	) => void;
}

export const AddDataSource: React.FC<IAddDataSourceProps> = ({
	className,
	close,
	groupId,
	open
}) => {
	const getDataSourceSections = () => [
		{
			dataSources: [
				{
					iconName: 'liferay-logo',
					iconSize: 'xxxl',
					name: Liferay.Language.get('liferay-dxp'),
					onClick: () => {
						analytics.track(
							'Clicked Create DXP Data Source - TEST',
							null,
							{
								ip: '0'
							}
						);

						open(modalTypes.CONNECT_DXP_MODAL, {
							groupId,
							onClose: close
						});
					},
					subtitle: Liferay.Language.get('supports-liferay-70+')
				},
				ENABLE_SALESFORCE && {
					iconName: 'salesforce-logo',
					iconSize: 'xxl',
					name: Liferay.Language.get('salesforce'),
					subtitle: Liferay.Language.get(
						'import-contacts-accounts-&-leads'
					),
					url: toRoute(Routes.SETTINGS_SALESFORCE_ADD, {groupId})
				},
				ENABLE_CSVFILE && {
					iconName: 'csv-logo',
					iconSize: 'xl',
					name: Liferay.Language.get('csv-file'),
					subtitle: Liferay.Language.get(
						'import-contacts-from-a-csv'
					),
					url: toRoute(Routes.SETTINGS_CSV_UPLOAD, {groupId})
				}
			].filter(Boolean) as [],
			title: Liferay.Language.get('choose-source')
		}
	];

	return (
		<BasePage
			breadcrumbItems={[
				breadcrumbs.getDataSources({groupId}),
				{
					active: true,
					label: Liferay.Language.get('add-data-source')
				}
			]}
			className={getCN('add-data-source-root', className)}
			groupId={groupId}
			pageDescription={Liferay.Language.get(
				'add-a-data-source-description'
			)}
			pageTitle={Liferay.Language.get('add-data-source')}
		>
			<SelectDataSource sections={getDataSourceSections()} />
		</BasePage>
	);
};

export default compose(
	withAdminPermission,
	connect(null, {close, open})
)(AddDataSource);
