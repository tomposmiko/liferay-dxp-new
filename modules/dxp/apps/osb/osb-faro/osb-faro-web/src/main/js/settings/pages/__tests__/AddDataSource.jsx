import mockStore from 'test/mock-store';
import React from 'react';
import {AddDataSource} from '../AddDataSource';
import {cleanup, render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {StaticRouter} from 'react-router-dom';

jest.unmock('react-dom');

describe('AddDataSource', () => {
	afterEach(cleanup);

	it('should render', () => {
		const {container} = render(
			<Provider store={mockStore()}>
				<StaticRouter>
					<AddDataSource groupId='23' />
				</StaticRouter>
			</Provider>
		);

		expect(container).toMatchSnapshot();
	});
});
