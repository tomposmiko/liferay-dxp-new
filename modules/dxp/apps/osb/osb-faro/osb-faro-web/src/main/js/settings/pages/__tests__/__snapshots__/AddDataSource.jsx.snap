// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDataSource should render 1`] = `
<div>
  <div
    class="settings-base-page-root add-data-source-root"
  >
    <div
      class="maintenance-alert-root"
    />
    <ol
      class="breadcrumb breadcrumbs-root"
    >
      <li
        class="breadcrumb-item"
      >
        <a
          class="breadcrumb-link breadcrumb-text-truncate"
          href="/workspace/23/settings/data-source"
        >
          Data Sources
        </a>
      </li>
      <li
        class="breadcrumb-item active"
      >
        <span
          class="breadcrumb-text-truncate"
        >
          Add Data Source
        </span>
      </li>
    </ol>
    <div
      class="content-header"
    >
      <div
        class="header-text"
      >
        <div
          class="d-flex"
        >
          <h3
            class="title-text text-truncate"
          >
            <span
              class="text-truncate"
              data-tooltip="false"
              title=""
            >
              Add Data Source
            </span>
          </h3>
        </div>
        <div
          class="description"
        >
          Add a Data Source to bring in data from various sources in to Analytics Cloud. Analytics Cloud automatic integration is supported for Liferay Portal version 7.0+. You can connect older versions of Liferay Portal by extracting data to a CSV and importing it as a CSV data source.
        </div>
      </div>
      <div
        class="page-actions-container"
      />
    </div>
    <div>
      <div
        class="select-data-source-root"
      >
        <section>
          <h4
            class="text-uppercase section-title"
          >
            Choose Source
          </h4>
          <div
            class="section-items"
          >
            <button
              class="button-root data-source-item btn btn-unstyled"
              type="button"
            >
              <div
                class="image"
              >
                <svg
                  class="lexicon-icon lexicon-icon-liferay-logo icon-root icon-size-xxxl"
                  role="presentation"
                >
                  <use
                    xlink:href="#liferay-logo"
                  />
                </svg>
              </div>
              <div
                class="details"
              >
                <div
                  class="title"
                >
                  <h4>
                    Liferay DXP
                  </h4>
                </div>
                <div
                  class="subtitle"
                >
                  Supports Liferay 7.0+
                </div>
              </div>
            </button>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
